### ISSUE TITLE
Wmbusmeters works with my raspberry but not on my synology using Docker

### ISSUE THREAD

**Issue by gd35** [weetmuts/wmbusmeters#650]
I made my own nanoCUL USB dongle. It works perfectly well with my rpi4 (without docker) based on the latest version of wmbusmeters I got from GitHub. However, I plan to use this nanoCUL with my Synology using Docker. I solved the driver issue (lack of usb support with DSM 7.x) and was able to install the container and to configure it. However, it does not work. I got the following entries in the log file:

(wmbusmeters) version: 1.8.0-13-ga5f0783
(config) store meter files in: "/wmbusmeters_data/logs/meter_readings"
(config) using device: /dev/ttyUSB0:cul:38400:t1 
(config) number of meters: 0
(lookup) with file/hex "/dev/ttyUSB0" cul
(serial) check if /dev/ttyUSB0 can be accessed
(serial) tty /dev/ttyUSB0 can be accessed
(serial) EVENT thread interrupted
(serial) before config: /dev/ttyUSB0 speed(B38400) input() output() control(CLOCAL|CREAD|CSIZE) local() special_chars(115,212,92,99,54,213,97,14) modem(DTR|RTS)
(serial) after config:  /dev/ttyUSB0 speed(B38400) input() output() control(CLOCAL|CREAD|CSIZE) local() special_chars(115,212,92,99,16) modem(DTR|RTS)
(serialtty) opened /dev/ttyUSB0 fd 3 (detect cul)
(serial /dev/ttyUSB0) sent "560D0A"
(serial) EVENT thread interrupted
(serial) received binary ""
(cul) probe response ""
(serial /dev/ttyUSB0) sent "560D0A"
(serial) EVENT thread interrupted
(serial) received binary ""
(cul) probe response ""
(serial /dev/ttyUSB0) sent "560D0A"
(serial) EVENT thread interrupted
(serial) received binary "5620312E3637206E616E6F43554C3836385F723536380D0A"
(cul) probe response "V 1.67 nanoCUL868_r568<0D><0A>"
(serialtty) closed /dev/ttyUSB0 (detect cul)
(serial) EVENT thread interrupted
(cul) are you there? yes
(main) opening /dev/ttyUSB0:cul:38400:t1
Started config cul on /dev/ttyUSB0 listening on t1
(cul) on /dev/ttyUSB0
(serial) EVENT thread interrupted
(serial) before config: /dev/ttyUSB0 speed(B38400) input() output() control(CLOCAL|CREAD|CSIZE) local() special_chars(115,212,92,99,54,213,97,14) modem(DTR|RTS)
(serial) after config:  /dev/ttyUSB0 speed(B38400) input() output() control(CLOCAL|CREAD|CSIZE) local() special_chars(115,212,92,99,16) modem(DTR|RTS)
(serialtty) opened /dev/ttyUSB0 fd 4 (cul)
(main) regular reset of cul  on /dev/ttyUSB0 will happen every 82800 seconds
(cul) set link mode t
(serial /dev/ttyUSB0) sent "6272740D0A"
(serial) EVENT thread interrupted
(wmbus device) timeout request id 1
(cul) received ""(cul) setting link mode(s) t1 is not supported for this cul device!
(serial) stopping manager

**wmbusmeters states that my nanoCUL does not support t1 although it worked fine with the raspberry (without docker):**

(wmbusmeters) version: 1.9.0-76-gb9f3446
(config) using device: /dev/ttyUSB0:cul:38400:t1 
(config) number of meters: 0
(lookup) with file/hex "/dev/ttyUSB0" cul
(serial) check if /dev/ttyUSB0 can be accessed
(serial) tty /dev/ttyUSB0 can be accessed
(serial) EVENT thread interrupted
(serial) before config: /dev/ttyUSB0 speed(B9600) input(ICRNL|IXON) output(ONLCR|OPOST) control(CLOCAL|CREAD|CSIZE|HUPCL) local(ECHO|ECHOCTL|ECHOE|ECHOK|ECHOKE|ICANON|IEXTEN|ISIG) special_chars(3,28,127,21,4,1,17,19,26,18,15,23,22) modem(DTR|RTS)
(serial) after config:  /dev/ttyUSB0 speed(B38400) input() output() control(CLOCAL|CREAD|CSIZE) local() special_chars() modem(DTR|RTS)
(serialtty) opened /dev/ttyUSB0 fd 3 (detect cul)
(serial /dev/ttyUSB0) sent "560D0A"
(serial) EVENT thread interrupted
(serial) received binary ""
(cul) probe response ""
(serial /dev/ttyUSB0) sent "560D0A"
(serial) EVENT thread interrupted
(serial) received binary ""
(cul) probe response ""
(serial /dev/ttyUSB0) sent "560D0A"
(serial) EVENT thread interrupted
(serial) received binary "5620312E3637206E616E6F43554C3836385F723536380D0A"
(cul) probe response "V 1.67 nanoCUL868_r568<0D><0A>"
(serialtty) closed /dev/ttyUSB0 (detect cul)
(cul) are you there? yes
(main) opening /dev/ttyUSB0:cul:38400:t1
Started config cul on /dev/ttyUSB0 listening on t1
(cul) on /dev/ttyUSB0
(serial) EVENT thread interrupted
(serial) before config: /dev/ttyUSB0 speed(B38400) input() output() control(CLOCAL|CREAD|CSIZE) local() special_chars() modem(DTR|RTS)
(serial) after config:  /dev/ttyUSB0 speed(B38400) input() output() control(CLOCAL|CREAD|CSIZE) local() special_chars() modem(DTR|RTS)
(serialtty) opened /dev/ttyUSB0 fd 3 (cul)
(main) regular reset of cul  on /dev/ttyUSB0 will happen every 82800 seconds
(cul) set link mode t
(serial /dev/ttyUSB0) sent "6272740D0A"
(serial) EVENT thread interrupted
(serial) received binary "544D4F44450D0A"
(cul) checkCULFrame "TMODE<0D><0A>"
(cul) no leading 'b' so it is text and no frame
(cul) received "TMODE"(serial /dev/ttyUSB0) sent "5832310D0A"
(wmbus) no alarm (expected activity) for cul
(serial) registered regular callback HOT_PLUG_DETECTOR(0) every 2 seconds
No meters configured. Printing id:s of all telegrams heard!
(serial) waiting for stop
(serial) EVENT thread interrupted
(serial) received binary "6231393434333034433431344143423230393830314646393841323531324230"
(cul) checkCULFrame "b1944304C414ACB209801FF98A2512B0"
(cul) no eol found yet, partial frame
(serial) received binary "3031334542313432353342324437304138303231383431334231334537383246"
(cul) checkCULFrame "b1944304C414ACB209801FF98A2512B0013EB14253B2D70A80218413B13E782F"
(cul) no eol found yet, partial frame
(serial) received binary "380D0A"
(cul) checkCULFrame "b1944304C414ACB209801FF98A2512B0013EB14253B2D70A80218413B13E782F8<0D><0A>"
(cul) checkCULFrame RSSI_RAW=-8
(cul) checkCULFrame LQI=65
(wmbus) trimming frame A "1944304C414ACB209801FF98A2512B0013EB14253B2D70A80218413B13E7"
(wmbus) ff a dll crc 0-9 ff98 ok
(wmbus) ff a dll crc mid 12-27 13e7 ok
(wmbus) trimming frame A "1944304C414ACB209801FF98A2512B0013EB14253B2D70A80218413B13E7"
(wmbus) trimmed 4 dll crc bytes from frame a and ignored 0 suffix bytes.
(wmbus) trimmed frame A "1944304C414ACB209801A2512B0013EB14253B2D70A80218413B"
(cul) received full T1 frame
(meter) no meter handled  checking 0 templates.
(diehl) preprocess necessary SAP_PRIOS
(diehl) Pre-processing: setting device type to water meter for SAP PRIOS
(wmbus) parseDLL @0 26
(wmbus) parseELL @10 16
(wmbus) parseNWL @10 16
(wmbus) parseAFL @10 16
(wmbus) parseTPL @10 16
(diehl) preprocess necessary SAP_PRIOS
(diehl) Pre-processing: setting device type to water meter for SAP PRIOS
(wmbus) parseDLL @0 26
(telegram) DLL L=19 C=44 (from meter SND_NR) M=4c30 (SAP) A=20cb4a41 VER=00 TYPE=07 (Water meter) (driver izar) DEV=cul RSSI=-78
(wmbus) parseELL @10 16
(wmbus) parseNWL @10 16
(wmbus) parseAFL @10 16
(wmbus) parseTPL @10 16
(telegram) TPL CI=a2
Received telegram from: 20cb4a41
          manufacturer: (SAP) Sappel (0x4c30)
                  type: Water meter (0x07)
                   ver: 0x00
                device: cul
                  rssi: -78 dBm
                driver: izar
(wmbus) 000   : 19 length (25 bytes)
(wmbus) 001   : 44 dll-c (from meter SND_NR)
(wmbus) 002   : 304c dll-mfct (SAP)
(wmbus) 004   : 414acb20 dll-id (20cb4a41)
(wmbus) 008   : 00 dll-version
(wmbus) 009   : 07 dll-type (Water meter)
(wmbus) 010   : a2 tpl-ci-field (Mfct specific)
(wmbus) 011 C?: 512B0013EB14253B2D70A80218413B mfct specific
(wmbus) telegram from 20cb4a41 ignored by all configured meters!

**I tried wmbusmeters on my Mac (without Docker) and got a similar issue:**

(wmbusmeters) version: 1.9.0-76-gb9f3446
(config) using device: /dev/cu.usbserial-1440:cul:38400:t1 
(config) number of meters: 0
(lookup) with file/hex "/dev/cu.usbserial-1440" cul
(serial) check if /dev/cu.usbserial-1440 can be accessed
(serial) tty /dev/cu.usbserial-1440 can be accessed
(serial) EVENT thread interrupted
(serial) before config: /dev/cu.usbserial-1440 speed(B9600) input() output() control(CREAD|CSIZE|HUPCL) local() special_chars(4,255,255,127,23,21,18,255,3,28,26,25,17,19,22,15,1,20,255) modem(DTR|RTS)
(serial) after config:  /dev/cu.usbserial-1440 speed(B38400) input() output() control(CLOCAL|CREAD|CSIZE) local() special_chars() modem(DTR|RTS)
(serialtty) opened /dev/cu.usbserial-1440 fd 3 (detect cul)
(serial /dev/cu.usbserial-1440) sent "560D0A"
(serial) EVENT thread interrupted
(serial) received binary ""
(cul) probe response ""
(serial /dev/cu.usbserial-1440) sent "560D0A"
(serial) EVENT thread interrupted
(serial) received binary ""
(cul) probe response ""
(serial /dev/cu.usbserial-1440) sent "560D0A"
(serial) EVENT thread interrupted
(serial) received binary "5620312E3637206E616E6F43554C3836385F723536380D0A"
(cul) probe response "V 1.67 nanoCUL868_r568<0D><0A>"
(serialtty) closed /dev/cu.usbserial-1440 (detect cul)
(cul) are you there? yes
(main) opening /dev/cu.usbserial-1440:cul:38400:t1
Started config cul on /dev/cu.usbserial-1440 listening on t1
(serial) EVENT thread interrupted
(cul) on /dev/cu.usbserial-1440
(serial) EVENT thread interrupted
(serial) before config: /dev/cu.usbserial-1440 speed(B9600) input() output() control(CREAD|CSIZE|HUPCL) local() special_chars(4,255,255,127,23,21,18,255,3,28,26,25,17,19,22,15,1,20,255) modem(DTR|RTS)
(serial) after config:  /dev/cu.usbserial-1440 speed(B38400) input() output() control(CLOCAL|CREAD|CSIZE) local() special_chars() modem(DTR|RTS)
(serialtty) opened /dev/cu.usbserial-1440 fd 3 (cul)
(main) regular reset of cul  on /dev/cu.usbserial-1440 will happen every 82800 seconds
(cul) set link mode t
(serial /dev/cu.usbserial-1440) sent "6272740D0A"
(serial) EVENT thread interrupted
(wmbus device) timeout request id 1
(cul) received ""(cul) setting link mode(s) t1 is not supported for this cul device!
(serial) stopping manager

I will be happy to get some advises. 

Thanks in advance,

**Comment by BIBOLV** [comment]
First log is from old version, so first, I would suggest to upgrade to latest version and try again. If it does not work then try to install docker on raspi and check if all works then.
Looking at logs it is visible that initial communication with dongle is successful in all cases, but it fails when mode is being set, so try to set different mode to check if it will fail also.
If nothing helps - my last suggestion would be to purchase IMST dongle to replace cul.

**Comment by gd35** [comment]
Thanks for the advice. Just after sending my first message, I realized that the official docker container had an older version of wmbusmeters. So I generated my own container with the latest version. The issue remains. So I will test a docker container with my rpi4 to check whether it is a docker issue. I will also test another mode. I recently found an unused rpi3b that I can use to connect to my homemade cul. It will be a cheaper option compared to the buying of an IMST dongle.
